feat: Updated LineChart in UI Kit [WEB-1700]

mapmeld opened the pull request:
## Description

Implementing three changes from the ticket:
- ~`ClusterHistoricalUsageChart` to use `LineChart` instead of directly accessing `UPlot`~
- `MetricType` is removed from `Serie`; display names of series are passed directly to make `LineChart` a more generic component
- `experimentId` prop is removed from `LineChart`; it was only passed by Profiler
~`import { ParallelCoordinates } from 'components/kit/LineChart'`~

## Test Plan

- Variable names remain logical and descriptive
~- While logged in as admin, view cluster history at `/det/clusters/historical-usage?after=2023-09-20&before=2023-10-04` and compare to latest-main original charts~
~- Reasonable toggle between dates on `/det/resourcepool/compute-pool/stats`~
- Profile charts appear correctly and can be downloaded from `/det/experiments/1788/profiler`

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

keita-determined reviewed (commented):
we wanna show all range (7 days / 30 days) in this stat charts instead of the minimum range. Ref: #7754

<img width="1464" alt="image" src="https://github.com/determined-ai/determined/assets/109113616/011ccabc-9dff-4200-ab43-6778d1256631">

--- 

the legend is gone. I think legend should be there

<img width="1451" alt="image" src="https://github.com/determined-ai/determined/assets/109113616/26e84c17-9fb4-47b9-bbb6-f3ac79b82a15">

mapmeld commented:
I've updated the `LineChart` component to now accept an `xRange` prop from that chart.
I added an example to `/det/design`

Also activated `showLegend` prop for `ClusterHistoricalUsageChart`

keita-determined reviewed (commented):
`Day` and `Month` range doesnt change
 
https://github.com/determined-ai/determined/assets/109113616/86e8760c-dde4-489f-9ad8-1154d7201d3a

keita-determined reviewed (commented):
bug: with single data point, the range doesnt look right. 

<img width="1473" alt="image" src="https://github.com/determined-ai/determined/assets/109113616/b8d946fd-c52f-4557-ac67-37f3b91ac1fd">

---

bug: in historical usage, charts are not synced

--- 

question:

the legend used to be selectable. Dont we wanna support it?

<img width="1454" alt="image" src="https://github.com/determined-ai/determined/assets/109113616/1553a6a9-d504-4ae8-a5d2-6c63297d6192">

mapmeld commented:
> bug: with single data point, the range doesnt look right.

For this one, the issue was `xTickValues` showing hh:mm:ss for such a short range in the time data. I modified the code so to use `xRange` for the range if it exists.

mapmeld commented:
can we make hovering/clicking chart legend -> interaction or focus on the chart , into a separate feature ticket?

keita-determined commented:
> can we make hovering/clicking chart legend -> interaction or focus on the chart , into a separate feature ticket?

I suggest you to ask about this in web channel since this can be a temporal feature regressions

keita-determined commented on webui/react/src/pages/F_ExpList/CompareMetrics.tsx:
y axis label is chopped. is it a blocker?

<img width="1014" alt="image" src="https://github.com/determined-ai/determined/assets/109113616/fcbc2953-a50c-4d48-9106-3556c0616114">

mapmeld replied:
I would be surprised if the axis labels here changed from the current one, do you have a link to the experiment?

keita-determined replied:
its in the new experiment list. Actually it happens in latest-main so not from this PR. however, i think this should be fixed since axis label is not visible at all

<img width="1040" alt="image" src="https://github.com/determined-ai/determined/assets/109113616/a24d0a67-92e0-4e0e-b351-10f955118c71">

keita-determined commented on webui/react/src/pages/Cluster/ClusterHistoricalUsage.tsx:
regression: cannot zoom in
also in Avg Queue Time chart too

https://github.com/determined-ai/determined/assets/109113616/a7355276-2884-4e14-9d4d-300f843829be

mapmeld replied:
upgraded to use the fix in determined-ui 0.5.8

keita-determined commented on webui/react/src/pages/Cluster.module.scss:
```diff
@@ -0,0 +1,10 @@
+      box-shadow: none;
+    }
+    > div {
+      transform: translate(0) !important;
```
is that possible not to use `!important`?

mapmeld replied:
unfortunately this has to override an element-level style from antd.Pivot
copy from https://github.com/determined-ai/determined/pull/7941

keita-determined commented on webui/react/src/pages/DesignKit.tsx:
```diff
@@ -606,8 +606,7 @@ const ChartsSection: React.FC = () => {
     },
-    metricType: MetricType.Training,
-    name: 'Line',
+    name: 'training.Line',
```
question: why is it hardcoded instead of `MetricType.Training`?

mapmeld replied:
we're removing the `MetricType` type so that all metric-groups are equal, and `LineChart` can be used by other groups within HPE without our own logic

keita-determined reviewed (commented):
found some regressions

keita-determined reviewed (commented):
Getting close but still some issues.

nit (not a blocker):  it makes sense, but the ticks look kinda random in time label. if its not `time`, 

<img width="98" alt="image" src="https://github.com/determined-ai/determined/assets/109113616/34492992-e7ce-4eb3-a2a8-79f955d5a27c">

---

bug: data not shown

https://github.com/determined-ai/determined/assets/109113616/e820fd91-e2e2-4af3-b557-331842e26489

---

nit: in http://latest-main.determined.ai:8080/det/experiments/4885/visualization/learning-curve the Y label looks unreasonably small in log scale

<img width="944" alt="image" src="https://github.com/determined-ai/determined/assets/109113616/a49aeb29-1200-4a27-9325-9ac1b40a19f0">

---

nit (not a blocker): usability: even though the cursor is not on the line, it goes to the trial detail page by clicking somewhere blank. My intention was to reset plot by double clicking 

https://github.com/determined-ai/determined/assets/109113616/163cbc62-3e61-4852-9769-6c42016cb3a2

mapmeld commented:
If the bug or condition is already on latest-main, they should go into a new ticket. Determined-UI changed the API / args for LineChart so we should sync up with that.

keita-determined commented:
the bug i label not blocker are the ones on latest-main. That's why I labeled not blocker. Other ones are from this PR

mapmeld commented:
I'll make a new ticket to get the cluster historical chart changes in a future PR. This might need to be solved with new ChartGrid opts, or some intensive debugging about how changing the fixed time x-axis and data affects the charts
